### ISSUE TITLE
 MA0124 misses array information from LoggerParameterTypes.txt

### ISSUE THREAD

**Issue by FreeYourMinds** [meziantou/Meziantou.Analyzer#623]
### Version of the Meziantou.Analyzer NuGet package

2.0.96

### Rule Identifier

MA0124

### Target Framework

net6.0,net7.0

### C# Language version

C# 11

### Description

# MA0124 missing array information

When using arrays as parameters to structured logging, the analyzer only picks up the base type discarding the array information:

## LoggerParameterTypes.txt

Matching the example code in `Reproduction Steps`
```
@PathParts;T:System.String[]
``` 

but MA0124 still reports 

> Error MA0124 : Parameter '@PathParts' must be of type 'string' but is of type 'string[]'

_Note: severity was intentially set to error_

### Reproduction Steps

Minimal code:

```c#
    private string EnsurePathExists(params string[] parts)
    {
        try
        {
            var path = Path.Combine(parts);
            var directoryInfo = Directory.CreateDirectory(path);

            return directoryInfo.FullName;
        }
        catch (Exception ex)
        {
            this.logger.LogWarning(ex, "Could not ensure path exists - Parts: '{@PathParts}': {Message}", parts, ex.Message);
            throw;
        }
    }
```

### Other information

_Off Topic (kind of)_

`MA0125` fails to recognize types that come from referenced NuGet packages, but guessing from my interpretation of the documentation this is by design, right?

**Comment by meziantou** [comment]
> MA0125 fails to recognize types that come from referenced NuGet packages, but guessing from my interpretation of the documentation this is by design, right?

Can you open another issue with a repro?

**Comment by meziantou** [comment]
Arrays are supported in the latest version. You can use

````
@PathParts;System.String[]
````

**Comment by FreeYourMinds** [comment]
> Arrays are supported in the latest version. You can use
> 
> ```
> @PathParts;System.String[]
> ```

Perfect, thank you for the quick fix. Working as expected now 👍

**Comment by FreeYourMinds** [comment]
Ok, after digging deeper for a minimal repro it turned out it was my mistake. I used a solution wide props file to add the LoggerParameterTypes.txt file, so MA0125 reporting

> Warning MA0125 : The type 'T:xxx' does not match any symbol of the compilation

was totally legit. The reported type came from a NuGet package that was not a part of all projects in the solution. 

So no bug, only human error here 🙈
